pkg/cmd/stack/translate.go

```diff
@@ -436,20 +437,21 @@ func translateServiceEnvironment(svc *model.Service) []apiv1.EnvVar {
 func translateContainerPorts(svc *model.Service) []apiv1.ContainerPort {
 	result := []apiv1.ContainerPort{}
 	for _, p := range svc.Ports {
-		result = append(result, apiv1.ContainerPort{ContainerPort: p})
+		result = append(result, apiv1.ContainerPort{ContainerPort: p.Port})
 	}
 	return result
 }
 
-func translateServicePorts(svc *model.Service) []apiv1.ServicePort {
+func translateServicePorts(svc model.Service) []apiv1.ServicePort {
 	result := []apiv1.ServicePort{}
 	for _, p := range svc.Ports {
```

Comment:
at the stack level, ports should be a list of integers? or do you want to support `protocol` in stacks too?